File: DataStructures/Arrays/Arrays.py
class Arrays:

    def findElement(self, key):
        for i in range(n):
            if arr[i] == key:
                return i
        return -1

    def insertElement(self, num):
        print("Array before insertion")
        print(arr)

        arr[5] = num

        print("Array after insertion")
        print(arr)

    def deleteElement(self, num):
        print("Array before deletion")
        print(arr)

        pos = obj.findElement(num)

        if pos == -1:
            print("Element not found")
            return

        for i in range(pos, len(arr) - 1):
            arr[i] = arr[i + 1]

        arr[len(arr) - 1] = 0

        print("Array after deletion")
        print(arr)


obj = Arrays()
arr = [0] * 6
arr[0] = 4
arr[1] = 3
arr[2] = 6
arr[3] = 1
arr[4] = 8
n = len(arr)

print(arr)

key = input("Enter a num to find in an array")

index = obj.findElement(int(key))

if index == -1:
    print("Element not found")
else:
    print("Element found at position " + str(index + 1))

num1 = input("Enter a num to insert in an array")

obj.insertElement(int(num1))

num2 = input("Enter a num to delete from an array")

obj.deleteElement(int(num2))

File: DataStructures/Arrays/Arrays.java
package DataStructures.Arrays;

import java.util.Scanner;

// Java program to implement arrays
public class Arrays {

    static int[] arr;
    static int intArr[];

    public static void main(String[] args) {
        arr = new int[6];
        arr[0] = 0;
        arr[1] = 1;
        arr[2] = 2;
        arr[3] = 3;
        arr[4] = 4;

        for (int i : arr) {
            System.out.println(i);
        }

        int n = 5, pos, x;
        intArr = new int[n + 1];
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter an array elements : ");
        for (int i = 0; i < n; i++) {
            intArr[i] = sc.nextInt();
        }

        for (int i : intArr) {
            System.out.println(i);
        }

        // Find an element in an array
        Scanner sc1 = new Scanner(System.in);
        System.out.println("Enter an element to find in an array : ");
        int position = findElement(intArr, sc1.nextInt());

        if (position == -1) {
            System.out.println("Element not found");
        } else {
            System.out.println("Element found at position " + (position + 1));
        }

        // Insert element at end of an array
        Scanner sc3 = new Scanner(System.in);
        System.out.println("Enter an element to insert in an array : ");
        insert(sc3.nextInt());


        // Delete an element from an array
        Scanner sc4 = new Scanner(System.in);
        System.out.println("Enter an element to delete in an array : ");
        int result = delete(sc4.nextInt());

        System.out.println("");

        if (result == 1) {
            System.out.println("Array after deletion");
            for (int i = 0; i < intArr.length -1; i++){
                System.out.print(intArr[i] + " ");
            }
        }

        System.out.println("");
        System.out.println(java.util.Arrays.toString(intArr));

    }

    /**
     * @param arr
     * @param num
     * @return index
     */
    public static int findElement(int arr[], int num) {
        for (int i = 0; i < intArr.length; i++) {
            if (intArr[i] == num) {
                return i;
            }
        }
        return -1;
    }

    /**
     * @param num
     * @return boolean
     */
    public static void insert(int num) {

        System.out.println("Array before inserting");
        System.out.println(java.util.Arrays.toString(intArr));

        intArr[intArr.length - 1] = num;

        System.out.println("Array after inserting");
        System.out.println(java.util.Arrays.toString(intArr));

    }

    /**
     * @param num
     */
    public static int delete(int num) {

        System.out.println("Array before deletion");
        System.out.println(java.util.Arrays.toString(intArr));

        // Find the element to be deleted from an array
        int pos = findElement(intArr, num);

        if (pos == -1) {
            System.out.println("Element not found");
            return -1;
        }

        int i;
        for (i = pos; i < intArr.length - 1; i++) {
            intArr[i] = intArr[i + 1];
        }

        // inserting last element with zero value
        intArr[intArr.length -1] = 0;

        return 1;
    }



}
